src/utilities/include/tri_dist.h

```diff
@@ -0,0 +1,225 @@
+      const float dd = glm::dot(V, V);
+
+      if (dd <= mindd) {
+        mindd = dd;
+
+        uint32_t id = i + 2;
+        if (id >= 3) id -= 3;
+        glm::vec3 Z = p[id] - cp;
+        float a = glm::dot(Z, V);
+        id = j + 2;
+        if (id >= 3) id -= 3;
+        Z = q[id] - cq;
+        float b = glm::dot(Z, V);
+
+        if ((a <= 0.0f) && (b >= 0.0f)) {
+          return glm::dot(V, V);
+        };
+
+        if (a <= 0.0f)
+          a = 0.0f;
+        else if (b > 0.0f)
+          b = 0.0f;
+
+        if ((mindd - a + b) > 0.0f) shown_disjoint = true;
+      }
+    }
+  }
+
+  glm::vec3 Sn = glm::cross(Sv[0], Sv[1]);
+  float Snl = glm::dot(Sn, Sn);
+
+  if (Snl > 1e-15f) {
+    const glm::vec3 Tp(glm::dot(p[0] - q[0], Sn), glm::dot(p[0] - q[1], Sn),
+                       glm::dot(p[0] - q[2], Sn));
+
+    int index = -1;
+    if ((Tp[0] > 0.0f) && (Tp[1] > 0.0f) && (Tp[2] > 0.0f)) {
+      index = Tp[0] < Tp[1] ? 0 : 1;
+      if (Tp[2] < Tp[index]) index = 2;
+    } else if ((Tp[0] < 0.0f) && (Tp[1] < 0.0f) && (Tp[2] < 0.0f)) {
+      index = Tp[0] > Tp[1] ? 0 : 1;
+      if (Tp[2] > Tp[index]) index = 2;
+    }
+
+    if (index >= 0) {
+      shown_disjoint = true;
+
+      const glm::vec3& qIndex = q[index];
+
+      glm::vec3 V = qIndex - p[0];
+      glm::vec3 Z = glm::cross(Sn, Sv[0]);
+      if (glm::dot(V, Z) > 0.0f) {
+        V = qIndex - p[1];
+        Z = glm::cross(Sn, Sv[1]);
+        if (glm::dot(V, Z) > 0.0f) {
+          V = qIndex - p[2];
+          Z = glm::cross(Sn, Sv[2]);
+          if (glm::dot(V, Z) > 0.0f) {
+            glm::vec3 cp = qIndex + Sn * Tp[index] / Snl;
+            glm::vec3 cq = qIndex;
+            return glm::dot(cp - cq, cp - cq);
+          }
+        }
+      }
+    }
+  }
+
+  glm::vec3 Tn = glm::cross(Tv[0], Tv[1]);
+  float Tnl = glm::dot(Tn, Tn);
+
+  if (Tnl > 1e-15f) {
+    const glm::vec3 Sp(glm::dot(q[0] - p[0], Tn), glm::dot(q[0] - p[1], Tn),
+                       glm::dot(q[0] - p[2], Tn));
+
+    int index = -1;
+    if ((Sp[0] > 0.0f) && (Sp[1] > 0.0f) && (Sp[2] > 0.0f)) {
+      index = Sp[0] < Sp[1] ? 0 : 1;
+      if (Sp[2] < Sp[index]) index = 2;
+    } else if ((Sp[0] < 0.0f) && (Sp[1] < 0.0f) && (Sp[2] < 0.0f)) {
+      index = Sp[0] > Sp[1] ? 0 : 1;
+      if (Sp[2] > Sp[index]) index = 2;
+    }
+
+    if (index >= 0) {
+      shown_disjoint = true;
+
+      const glm::vec3& pIndex = p[index];
+
+      glm::vec3 V = pIndex - q[0];
+      glm::vec3 Z = glm::cross(Tn, Tv[0]);
+      if (glm::dot(V, Z) > 0.0f) {
+        V = pIndex - q[1];
+        Z = glm::cross(Tn, Tv[1]);
+        if (glm::dot(V, Z) > 0.0f) {
+          V = pIndex - q[2];
+          Z = glm::cross(Tn, Tv[2]);
+          if (glm::dot(V, Z) > 0.0f) {
+            glm::vec3 cp = pIndex;
+            glm::vec3 cq = pIndex + Tn * Sp[index] / Tnl;
+            return glm::dot(cp - cq, cp - cq);
+          }
+        }
+      }
+    }
+  }
+
+  return shown_disjoint ? mindd : 0.0f;
```

Comment:
This is where I find code coverage to be a useful check - it's a little surprising that execution never reaches this point. I wouldn't consider this a blocker, but it's interesting - sometimes it means we just haven't picked enough variety in our tests, but sometimes it points to vestigial code that actually can't run. It might be worth throwing in a larger, more complicated test object  (using a large search radius) just to see if we can hit all the corners of this algorithm.

Reply:
I'll check it out!

Reply:
Added a test case with you stretchy bracelet design and it seems to cover almost everything!